### DISCUSSION TITLE
Problem building executable from source on OS X and Linux

### DISCUSSION THREAD

**Discussion by wagnar-woofbrok** [keepassxreboot/keepassxc#8509] [Q&A] [answered]
Hello,

I followed the directions for building keepassxc from source on OS X [listed here](https://github.com/keepassxreboot/keepassxc/blob/develop/INSTALL.md), but I'm having an issue where the process completes OK but I cannot find the executable. I tried the Linux instructions on a Debian flavor Linux machine and had the same issue.

The directions say to cd into the keepassxc directory after a git clone, checkout latest, and then run these commands:

mkdir -p build
cd build 
cmake -DWITH_XC_ALL=ON ..
make

Then the executable is supposed to be in the directory `./build/src`.

However, I cannot run the cmake command from `keepassxc/build/` because it complains that this directory "does not contain CMakeLists.txt" (which is true). If I run the commands from the `keepassxc/` directory, it completes OK but there is no `./build/src` directory.

Can someone please help me figure out what I'm misunderstanding or not seeing?

**Comment by droidmonkey** [comment, marked as answer]
You forgot the `..` at the end of the cmake command when you are in the build folder.

**Reply by michaelk83** [reply]
Wouldn't it be more intuitive to run `cmake` from the root directory like this user has, and have it take care of the `build` subfolder on its own (without all the extra `mkdir`, `cd`, `..`)?

**Reply by droidmonkey** [reply]
CMake philosophy is to be performed in an immutable way from your source folder. So that means starting from a folder YOU make.
